src/Type/Php/RangeFunctionReturnTypeExtension.php

```diff
@@ -112,6 +112,9 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		$isStepInteger = $stepType->isInteger()->yes();
 
 		if ($isInteger && $isStepInteger) {
+			if ($argType instanceof IntegerRangeType) {
+				return AccessoryArrayListType::intersectWith(new ArrayType($argType, $argType));
+			}
 			return AccessoryArrayListType::intersectWith(new ArrayType(new IntegerType(), new IntegerType()));
```

Comment:
can it be simplified to
```suggestion
			return AccessoryArrayListType::intersectWith(new ArrayType($argType, $argType));
```

Reply:
I did that originally, it won't work with scalar types, as you'll get a RT as `array{1,10}` for `range(1, 10)` (or similar)

Reply:
which also makes me wonder if the IntRange should be implied from scalar types :thinking: